File: app/src/main/java/yiyo/gitlabandroid/utils/Configuration.java
package yiyo.gitlabandroid.utils;

import android.content.Context;
import android.content.SharedPreferences;

/**
 * Created by yiyo on 24/07/15.
 */
public class Configuration {

    private SharedPreferences sharedPreferences;
    private SharedPreferences.Editor editor;
    private Context mContext;

    private static final String SHARED_PREFS_NAME = "GITLAB_ANDROID_PREFS";

    // Llaves del SharedPreferences
    private static final String IS_LOGGED_IN = "is_logged_in";
    private static final String NAME = "name";
    private static final String USERNAME = "username";
    private static final String EMAIL = "email";
    private static final String PRIVATE_TOKEN = "private_token";
    private static final String USER_ID = "user_id";

    public Configuration(Context mContext) {
        this.mContext = mContext;
        sharedPreferences = this.mContext.getSharedPreferences(SHARED_PREFS_NAME, Context.MODE_PRIVATE);
        editor = sharedPreferences.edit();
        editor.apply();
    }

    public boolean createSession(String name, String username, String email, String privateToken, Integer id) {
        if (privateToken != null) {
            editor.putString(NAME, name);
            editor.putString(USERNAME, username);
            editor.putString(EMAIL, email);
            editor.putString(PRIVATE_TOKEN, privateToken);
            editor.putInt(USER_ID, id);
            editor.putBoolean(IS_LOGGED_IN, true);
            return editor.commit();
        } else {
            return false;
        }
    }

    /**
     * Check for login state
     */
    public boolean isLoggedIn() {
        return sharedPreferences.getBoolean(IS_LOGGED_IN, false);
    }

    /**
     * Cleaning all data from Shared Preferences
     */
    public void closeSession() {
        editor.remove(NAME);
        editor.remove(USERNAME);
        editor.remove(EMAIL);
        editor.remove(PRIVATE_TOKEN);
        editor.remove(USER_ID);
        editor.putBoolean(IS_LOGGED_IN, false);
        editor.apply();
    }
}

File: app/src/main/kotlin/yiyo/gitlabandroid/utils/Configuration.kt
package yiyo.gitlabandroid.utils

import android.content.Context
import android.content.SharedPreferences

/**
* Created by yiyo on 24/07/15.
*/
class Configuration(val mContext: Context) {

    public val sharedPreferences: SharedPreferences
    private val editor: SharedPreferences.Editor

    companion object {
        // Nombre del SharedPreferences de la aplicación
        private val SHARED_PREFS_NAME = "GITLAB_ANDROID_PREFS"

        // Llaves del SharedPreferences
        private val IS_LOGGED_IN = "is_logged_in"
        val NAME = "name"
        val USERNAME = "username"
        private val EMAIL = "email"
        private val PRIVATE_TOKEN = "private_token"
        private val USER_ID = "user_id"
    }

    init {
        sharedPreferences = mContext.getSharedPreferences(SHARED_PREFS_NAME, Context.MODE_PRIVATE)
        editor = sharedPreferences.edit()
        editor.apply()
    }

    /**
     * Create session for a user
     */
    fun createSession(name: String, username: String, email: String, privateToken: String, id: Int): Boolean {
        editor.putString(NAME, name)
        editor.putString(USERNAME, username)
        editor.putString(EMAIL, email)
        editor.putString(PRIVATE_TOKEN, privateToken)
        editor.putInt(USER_ID, id)
        editor.putBoolean(IS_LOGGED_IN, true)
        return editor.commit()
    }

    /**
     * Check for login state
     */
    fun isLoggedIn() = sharedPreferences.getBoolean(IS_LOGGED_IN, false)

    /**
     * Cleaning all data from Shared Preferences
     */
    fun closeSession() {
        editor.remove(NAME)
        editor.remove(USERNAME)
        editor.remove(EMAIL)
        editor.remove(PRIVATE_TOKEN)
        editor.remove(USER_ID)
        editor.putBoolean(IS_LOGGED_IN, false)
        editor.apply()
    }

    fun getUserDetails() = hashMapOf(
            NAME  to sharedPreferences.getString(NAME, null),
            USERNAME to sharedPreferences.getString(USERNAME, null))

    fun getPrivateToken() = sharedPreferences.getString(PRIVATE_TOKEN, null)
}
